core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProvider.java

```diff
@@ -110,26 +114,20 @@ public DirectoryBackedArtifactSignerProvider(
   }
 
   public void cacheAllSigners() {
-    final Set<String> identifiers = allIdentifiers();
-    LOG.info("Loading {} signers", identifiers.size());
-    identifiers.parallelStream().forEach(this::cacheSigner);
+    final Collection<ArtifactSigner> allSigners =
+        findSigners(this::matchesFileExtension, signerParser::parse);
+    allSigners.forEach(
+        signer -> artifactSignerCache.put(normaliseIdentifier(signer.getIdentifier()), signer));
```

Comment:
allIdentifiers does this the same way I think, so could probably moved allSigners into a single function that both can use. Then allIdentifiers can pull the identifiers out and this can just add them to the cache.